src/cascadia/Remoting/Monarch.cpp

```diff
@@ -574,35 +574,61 @@ namespace winrt::Microsoft::Terminal::Remoting::implementation
-    //   themselves.
-    // - This will tell each and every peasant to identify themselves. This will
-    //   eventually propagate down to TerminalPage::IdentifyWindow.
+    // - Helper for doing something on each and every peasant, with no regard
+    //   for if the peasant is living or dead.
+    // - We'll try calling callback on every peasant.
+    // - If any single peasant is dead, then we'll call errorCallback, and move on.
+    // - We're taking an errorCallback here, because the thing we usually want
+    //   to do is TraceLog a message, but TraceLoggingWrite is actually a macro
+    //   that _requires_ the second arg to be a string literal. It can't just be
+    //   a variable.
     // Arguments:
-    // - <unused>
+    // - callback: The function to call on each peasant
+    // - errorCallback: The function to call if a peasant is dead.
     // Return Value:
     // - <none>
-    void Monarch::_identifyWindows(const winrt::Windows::Foundation::IInspectable& /*sender*/,
-                                   const winrt::Windows::Foundation::IInspectable& /*args*/)
+    void Monarch::_forAllPeasantsIgnoringTheDead(std::function<void(const Remoting::IPeasant&, const uint64_t)> callback,
+                                                 std::function<void(const uint64_t)> errorCallback)
     {
-        // Notify all the peasants to display their ID.
         for (const auto& [id, p] : _peasants)
         {
             try
             {
-                p.DisplayWindowId();
+                callback(p, id);
             }
             catch (...)
             {
                 LOG_CAUGHT_EXCEPTION();
                 // If this fails, we don't _really_ care. Just move on to the
                 // next one. Someone else will clean up the dead peasant.
-                TraceLoggingWrite(g_hRemotingProvider,
-                                  "Monarch_identifyWindows_Failed",
-                                  TraceLoggingInt64(id, "peasantID", "The ID of the peasant which we could not identify"),
-                                  TraceLoggingLevel(WINEVENT_LEVEL_VERBOSE));
+                errorCallback(id);
             }
         }
     }
+
+    // Method Description:
+    // - This is an event handler for the IdentifyWindowsRequested event. A
+    //   Peasant may raise that event if they want _all_ windows to identify
+    //   themselves.
+    // - This will tell each and every peasant to identify themselves. This will
+    //   eventually propagate down to TerminalPage::IdentifyWindow.
+    // Arguments:
+    // - <unused>
+    // Return Value:
+    // - <none>
+    void Monarch::_identifyWindows(const winrt::Windows::Foundation::IInspectable& /*sender*/,
+                                   const winrt::Windows::Foundation::IInspectable& /*args*/)
+    {
+        // Notify all the peasants to display their ID.
+        auto callback = [](auto&& p, auto&& /*id*/) {
+            p.DisplayWindowId();
+        };

```

Comment:
Wait... if we're not using `id`, should we just remove it from the first param in `_forAllPeasantsIgnoringTheDead`?

Reply:
Mike: writes generic function

Reviewers: _is it possibly too generic?_

Reply:
Meh, I presumed a future user of this helper might want it.